hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,282 @@
+    node_handle_.getParam("base_link",config_.base_link) &&
+    node_handle_.getParam("world_frame",config_.world_frame) &&
+    node_handle_.getParam("controller_joint_names",config_.joint_names) &&
+    node_handle_.getParam("moveit_params/plan_time", config_.plan_time) &&
+    node_handle_.getParam("moveit_params/step_size", config_.step_size) &&
+    node_handle_.getParam("moveit_params/jump_thresh", config_.jump_thresh) &&
+    node_handle_.getParam("moveit_params/replan", config_.moveit_replan) &&
+    node_handle_.getParam("moveit_params/avoid_collision", config_.avoid_collision) &&
+    node_handle_.getParam("moveit_params/moveit_planner_type",config_.moveit_planner_type) &&
+    node_handle_.getParam("descartes_params/descartes_planner_type",config_.descartes_planner_type) &&
+    node_handle_.getParam("trajectory/time_delay",config_.time_delay) &&
+    node_handle_.getParam("trajectory/time_offset",config_.time_offset))
+    {
+      ROS_INFO_STREAM("Loaded parameters");
+    }
+    else
+    {
+      ROS_ERROR_STREAM("Failed to load application parameters");
+      exit(-1); //TO CHECK
+    }
+
+    // Set the group name of HybridPlanner's MoveIt! group member in accordance to its config_ struct
+    moveit::planning_interface::MoveGroup local_group(config_.robot_description);
+    moveit_group_ = local_group;
+    
+    // Initialize a Descartes robot model, MoveIt group is already initialized to config_.group_name in HybridPlanner class // TOCHECK here v/s in class definition itself
+    if (!descartes_model_->initialize(config_.robot_description, config_.group_name, config_.world_frame, config_.tcp_frame))
+    {
+      ROS_WARN("Could not initialize Descartes robot model");
+    }
+
+    // Initialize a Descartes planner with the Descartes Model 
+    if(config_.descartes_planner_type == "dense")
+    {
+      descartes_planner_dense_.initialize(descartes_model_);
+      ROS_INFO("Initialized Descartes dense planner");
+    }
+    else if(config_.descartes_planner_type == "sparse")
+    {
+      descartes_planner_sparse_.initialize(descartes_model_);
+      ROS_INFO("Initialized Descartes sparse planner");
+    }
+    else
+    {
+      ROS_WARN("Descartes planner not initialized. Errors will occur if process path segments are being planned for");
+    }
+
+    // The overall_robot_traj_ is the final, hybrid trajectory that is meant to be executed.  
+    robot_trajectory::RobotTrajectory local_overall_robot_traj(local_group.getCurrentState()->getRobotModel(), config_.robot_description); //TODO group.getRobotModel
+    overall_robot_traj_ = local_overall_robot_traj;  
+  }
+
+  void HybridPlanner::appendFreeMotionTrajectorySegment(std::vector<geometry_msgs::Pose>& waypoints)
+  {
+    TrajectorySegment current_free_segment;
+    current_free_segment.type_ = FREE_MOTION;
+    current_free_segment.waypoints_geom_msgs_ = waypoints;
+    unplanned_trajectory_.push_back(current_free_segment);      
+  }
```

Comment:
I think that it'll be helpful to have this method return a boolean in order to inform if the waypoints were properly added to the path even though no error checking is being done right now. I envision that at some point, some data integrity checks could be made in order to ensure that the waypoints contain the right kind of information.